readthedocs/restapi/views/integrations.py

```diff
@@ -191,7 +212,13 @@ def handle_webhook(self):
         # Handle push events and trigger builds
         if event == GITLAB_PUSH:
             try:
-                branches = [self.request.data['ref'].replace('refs/heads/', '')]
+                # GitLab only returns one branch.
+                branch = {
+                    'name': self.request.data['ref'].replace('refs/heads/', ''),
+                    # Assuming the first element is the last commit.
+                    'last_commit': self.request.data['commits'][0],
+                }
+                branches = [branch]
```

Comment:
Another thought here... If we already know at this point the last commit message, shouldn't skip here the call to `self.get_response_push` instead of going deeper in the path?

Reply:
We need the (buid, not_building) from the `build_branches` function to make the response.